C++/include/algorithm/dynamic_programming/0_1_knapsack.hpp

```diff
@@ -0,0 +1,42 @@
+/*
+  0-1 Knapsack Problem
+  --------------------
+  Given weights and values of n items, put these items in a knapsack
+  of a fixed capacity to get the maximum total value in the knapsack.
+
+  Time complexity
+  ---------------
+  O(N*C), where N is the number of items and C is the capacity of the knapsack.
+
+  Space complexity
+  ----------------
+  O(N*C), the variables same as time complexity.
+
+  Author
+  ------
+  Manas Gupta (@heisenberg-2505)
+*/
+
+#ifndef KNAPSACK_HPP
+#define KNAPSACK_HPP
+#include <algorithm>
+#include <vector>
+using namespace std;
```

Comment:
[Avoid `using namespace std`](https://stackoverflow.com/questions/1452721/why-is-using-namespace-std-considered-bad-practice). Instead, reference what you need specifically, as in: 

`using std::vector;`